### ISSUE TITLE
Add link abstractions?

### ISSUE THREAD

**Issue by trev-dev** [jakewvincent/mkdnflow.nvim#30]
I enjoy that Vimwiki will simplify links to just their link names, then they expand to the [link name](./file.md) when the cursor is on top of them. It's just a nice feature that de-clutters the viewing experience.

**Comment by jakewvincent** [comment]
I agree that the concealing that Vimwiki and others do would be nice, especially since markdown-style links can look overly long. I'll try to look into this soon and will comment again here if/when I have something to be tested.

**Comment by trev-dev** [comment]
Thanks Jake. I would not mind taking a stab at this if you're open to it. I looked up `:help conceal` right after posting this and gave myself a big eye-roll. It may be an easy first issue.

**Comment by jakewvincent** [comment]
Yeah, go for it! Organizationally, I'm not totally sure where this should go since none of the Lua scripting so far relates to visuals. For easy testing you might want to place it in init.lua and then once you've got it working create a separate module that gets loaded in init.lua if a config option is set. Whatever makes sense to you.

**Comment by trev-dev** [comment]
Hey, apologies. I've taken a stab at this a couple of times in the last couple of weeks. Turns out grasping all the quirks of vim's `syntax` system is escaping me. I'm trying to change things with Lua, nothing seems to be happening. I have written syntax with vimscript in the past and it had worked. I have dumbed down the implementation to match a simple string, it won't conceal.

I am losing confidence in my configuration/set-up or perhaps just my ability to help with this in some timely manner.

**Comment by jakewvincent** [comment]
No worries! I haven't done anything with conceal before but would like to try to figure it out. I'll share details here if I figure it out just in case it's of interest.

**Comment by trev-dev** [comment]
It should be almost dead simple*

* provided that your syntax configuration is sound and you know your pearl regex.

Its supposed to be a simple match group:

```vimscript
syntax match MkdnflowHideable conceal ”foo"
set conceallevel 1 ; something larger than 0
```

Obviously syntax should be `on`. I think my problem is I am not using the ./syntax/markdown.vim path but this is Lua...so...what now?

**Comment by jakewvincent** [comment]
@trev-dev Just fiddling a little bit with this... you can still add a vimscript file under `mkdnflow.nvim/syntax/`. I did get a little bit of concealing to work when placed there.

**Comment by trev-dev** [comment]
I was beating my head against making it work from the `lua/` folder as a configuration import in the entrypoint (`init.conceal`).

**Comment by jakewvincent** [comment]
Update: Actually it looks like it can be placed in `lua`. If you have a file called `conceal.lua` and load it in the setup function (by adding `init.conceal = require('mkdnflow.conceal')` after `load_on_ft[ft]`, the above conceals will load properly. They have to be wrapped in lua functions of course:
```lua
vim.api.nvim_command("call matchadd('Conceal', '\\[\\ \\]', 0, 11, {'conceal': ''})")
vim.api.nvim_command("call matchadd('Conceal', '\\[x\\]', 0, 12, {'conceal': ''})")
vim.api.nvim_command("highlight Conceal ctermbg=NONE ctermfg=NONE guibg=NONE guifg=NONE")
vim.api.nvim_win_set_option(0, 'conceallevel', 1)
```

**Comment by jakewvincent** [comment]
If you're interested, you should continue! I probably wouldn't be able to work on this for a few more days. If you get it to the point where you're relatively happy with the conceals, you can submit a PR and I can go ahead and do the configuration to enable/disable the conceals in the user config.

**Comment by trev-dev** [comment]
I am! Would much rather own my original offer to help out than drop it. The collaborative effort has been a really nice touch. Thanks for stepping in with me.

**Comment by jakewvincent** [comment]
@trev-dev how goes it on the conceal front?

**Comment by trev-dev** [comment]
Apologies, and thanks for the reminder. I've moved on from using NeoVim for the time being and I lost track of this.

I took another stab at it regardless and pattern matching something like `[<stuff has to be here but don't include it>](possible/path/string)` is driving me a bit nuts. I'm going to have to pass the torch back.

**Comment by jakewvincent** [comment]
No worries, thanks for the update!

**Comment by jakewvincent** [comment]
Added in commit 23c22f10735f98e496d538decbc48159e4308952; currently testing in dev branch.
